plugin/src/main/java/com/denizenscript/denizen/scripts/commands/npc/AssignmentCommand.java

```diff
@@ -68,81 +65,55 @@ public AssignmentCommand() {
     //
     // @Usage
     // Use to remove an extra assignment from the NPC.
-    // - assignment add script:name_fix_assign
+    // - assignment remove script:name_fix_assign
     // -->
 
-    private enum Action {SET, ADD, REMOVE, CLEAR}
+    public enum Action { SET, ADD, REMOVE, CLEAR }
 
     @Override
     public void addCustomTabCompletions(TabCompletionsBuilder tab) {
-        tab.add("set", "add", "remove", "clear");
+        tab.add(Action.values());
         tab.addScriptsOfType(AssignmentScriptContainer.class);
     }
 
-    @Override
-    public void parseArgs(ScriptEntry scriptEntry) throws InvalidArgumentsException {
-        for (Argument arg : scriptEntry) {
-            if (arg.matchesEnum(Action.class)
-                    && !scriptEntry.hasObject("action")) {
-                scriptEntry.addObject("action", Action.valueOf(arg.getValue().toUpperCase()));
-            }
-            else if (arg.matchesArgumentType(ScriptTag.class)
-                    && !scriptEntry.hasObject("script")) {
-                // Check the type of script.. it must be an assignment-type container
-                if (arg.asType(ScriptTag.class) != null
-                        && arg.asType(ScriptTag.class).getContainer() instanceof AssignmentScriptContainer) {
-                    scriptEntry.addObject("script", arg.asType(ScriptTag.class));
-                }
-                else {
-                    throw new InvalidArgumentsException("Script specified is not an 'assignment-type' container.");
-                }
-            }
-            else if (!scriptEntry.hasObject("npcs")
-                    && arg.matchesArgumentList(NPCTag.class)) {
-                scriptEntry.addObject("npcs", arg.asType(ListTag.class).filter(NPCTag.class, scriptEntry));
-            }
-            else {
-                arg.reportUnhandled();
-            }
-        }
-        if (!scriptEntry.hasObject("npcs")) {
+    public static void autoExecute(ScriptEntry scriptEntry,
+                                   @ArgName("action") Action action,
+                                   @ArgName("script") @ArgPrefixed @ArgDefaultNull ScriptTag script,
+                                   @ArgName("to") @ArgPrefixed @ArgDefaultNull @ArgSubType(NPCTag.class) List<NPCTag> to) {

```

Comment:
The original code treated the prefix as optional, this is a potentially breaking change of outdated/no-longer-documented behavior. Offhand genuinely entirely can't remember if using it without the prefix would've been shown anywhere. Probably no? but still better to deprecation cycle it.